### ISSUE TITLE
Feature request: Support automatic login to Leta search

### ISSUE THREAD

**Issue by yanagibashi-mt** [mullvad/browser-extension#153]
When cookies are cleared in the browser, you get logged out of Mullvad Leta search. This can be quite a hassle, especially for a search engine. It would be nice if the browser extension could also help with automatically logging the user in for these situations. This is relevant for users who often use private browsing mode, or who use Mullvad Browser, or similar setups that clear site data frequently.

A nice example is the extension for Kagi search: https://github.com/kagisearch/browser_extensions When using it, you are always logged into Kagi and you are able to search.

**Comment by Crehet** [comment]
While the extension is a cool thing to have, I'd prefer for Mullvad Leta Search Engine to simply automatically recognize a Mullvad VPN user upon loading Mullvad Leta's page, similarly to how mullvad.net does it when performing a [connection check](https://mullvad.net/en/check).

Honestly I would like an explanation to why does a user already using Mullvad VPN even needs to login to Leta.

**Comment by begin-theadventure** [comment]
`Cookies and Site Data` -> `Manage Exceptions...`?

**Comment by yanagibashi-mt** [comment]
> Honestly I would like an explanation to why does a user already using Mullvad VPN even needs to login to Leta.

There is a limit on the amount of daily searches per account. And while the Leta servers can recognize a Mullvad user by their IP address, they obviously won't know which user account is being used just from the IP address. So that's why there's an additional login required in order to be able to enforce the daily search restriction.

**Comment by Crehet** [comment]
> `Cookies and Site Data` -> `Manage Exceptions...`?

It doesn't work for me on Firefox with a fresh profile and without Mullvad's extension. Once Firefox closes; I have to re-enter my credentials again.

**Comment by yanagibashi-mt** [comment]
> > `Cookies and Site Data` -> `Manage Exceptions...`?
> 
> It doesn't work for me on Firefox with a fresh profile and without Mullvad's extension. Once Firefox closes; I have to re-enter my credentials again.

Because the person who answered with that suggestion didn't read the OP fully. Exceptions are only an option in case your browser actually persists the exceptions themselves, which depending on your settings might not be the case.

Though with a vanilla Firefox without extensions and a fresh profile, these exceptions should not get cleared (and cookies also should not, for what it's worth). It might be the case that the cookie set by Leta simply expired, maybe because it's bound to one session, or the Leta servers require re-authentication every so often regardless of your saved cookies. I haven't looked into it that much to see if either one is the case.

In any way, there will always be many avenues that might lead to the user getting logged out. That's why having the extension handle it is imo one of the best options forward here, and it seems like the Mullvad team agrees.

**Comment by ruihildt** [comment]
> Honestly I would like an explanation to why does a user already using Mullvad VPN even needs to login to Leta.

See the FAQ: https://leta.mullvad.net/faq#why-is-there-a-limit
